src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java

```diff
@@ -181,10 +197,17 @@ private static Stream<LocalDateTime> nonWorkingHoursScenarioProvider() {
 
     private static Stream<LocalDateTime> workingHoursScenarioProvider() {
         return Stream.of(
-            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 9, 30),
-            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 9, 31),
-            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 18, 30),
-            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 18, 29)
+            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 8, 30),
+            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 8, 31),
+            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 16, 29),
+            LocalDateTime.of(2022, Month.SEPTEMBER, 01, 16, 30)
+        );
+    }
+
+    private static Stream<LocalDateTime> workingHoursWithTimeZoneScenarioProvider() {
+        return Stream.of(
+            LocalDateTime.of(2024, Month.JANUARY, 01, 10, 00),
+            LocalDateTime.of(2024, Month.MAY, 01, 9, 00)
```

Comment:
I think we may need more tests unless they exist already which is not in this PR?  Let me know if already covered elsewhere.

We want tests where the clock is at UTC/GMT date and the time is within working hours and outside working hours.  We'd want to test the beginning of the day and the end of the day.
Then we also want tests where we are in BST and the time is within working hours and outside working hours.  Again we'd want to test the beginning and end of the day for those.

Also there's a scenario where no messages are received during working hours?

I think here you are covering some of the positive cases but not all.

Reply:
The old unit tests in the ReceivedMessagesHealthControllerTest.java covered the scenarios mentioned in your comment Ravi.
Eg: 
1. test_health_does_not_call_repository_if_working_day_time_is_out_of_working_hours
2. test_health_calls_repository_if_working_day_time_is_within_working_hours
3. test_health_reports_error_if_no_messages_received_in_last_hour